src/python/gem5/components/processors/base_cpu_core.py

```diff
@@ -155,9 +155,9 @@ def get_mmu(self) -> BaseMMU:
     @overrides(AbstractCore)
     def set_simpoint(self, inst_starts: List[int], init: bool) -> None:
         if init:
-            self.core.simpoint_start_insts = inst_starts
+            self.core.simpoint_start_insts = [*set(inst_starts)]
```

Comment:
I'm not sure there's any guarantee that this is in order. It does seem to be in practice, but I can't find where in the documentation it is guaranteed. 

```suggestion
            self.core.simpoint_start_insts = sorted(set(inst_starts))
```

Reply:
Yes, you're right. I changed it to using sorted().